packages/analytics-controller/src/AnalyticsController.test.ts

```diff
@@ -214,239 +486,413 @@ describe('AnalyticsController', () => {
+          optedIn: true,
+          analyticsId: 'ffffffff-ffff-4fff-8fff-ffffffffffff',
+        },
         platformAdapter: mockAdapter,
       });
 
-      messenger.call('AnalyticsController:identify', 'user-123', {
-        email: 'test',
-      });
+      controller.trackView('home', { referrer: 'test' });
 
-      expect(mockAdapter.identify).toHaveBeenCalled();
-      expect(controller.state.analyticsId).toBe('user-123');
+      expect(mockAdapter.view).toHaveBeenCalledWith('home', {
+        referrer: 'test',
+      });
     });
 
-    it('handles trackPage action', () => {
+    it('does not call platform adapter when disabled', async () => {
       const mockAdapter = createMockAdapter();
-      const { messenger } = setupController({ platformAdapter: mockAdapter });
+      const { controller } = await setupController({
+        state: {
+          optedIn: false,
+          analyticsId: 'abcdef01-2345-4678-9abc-def012345678',
+        },
+        platformAdapter: mockAdapter,
+      });
 
-      messenger.call('AnalyticsController:trackPage', 'home', {});
+      controller.trackView('home');
 
-      expect(mockAdapter.trackPage).toHaveBeenCalled();
+      expect(mockAdapter.view).not.toHaveBeenCalled();
     });
+  });
 
-    it('handles enable action', () => {
-      const { controller, messenger } = setupController({
-        state: { enabled: false },
+  describe('optIn', () => {
+    it('sets optedIn to true', async () => {
+      const { controller } = await setupController({
+        state: {
+          ...getDefaultAnalyticsControllerState(),
+          analyticsId: 'fedcba98-7654-4321-8fed-cba987654321',
+        },
       });
 
-      expect(controller.state.enabled).toBe(false);
-
-      messenger.call('AnalyticsController:enable');
+      controller.optIn();
 
-      expect(controller.state.enabled).toBe(true);
+      expect(controller.state.optedIn).toBe(true);
     });
+  });
 
-    it('handles disable action', () => {
-      const { controller, messenger } = setupController({
-        state: { enabled: true },
+  describe('optOut', () => {
+    it('sets optedIn to false', async () => {
+      const { controller } = await setupController({
+        state: {
+          optedIn: true,
+          analyticsId: '01234567-89ab-4cde-8f01-23456789abcd',
+        },
       });
 
-      expect(controller.state.enabled).toBe(true);
-
-      messenger.call('AnalyticsController:disable');
-
-      expect(controller.state.enabled).toBe(false);
-    });
-
-    it('handles optIn action', () => {
-      const { controller, messenger } = setupController();
+      controller.optOut();
 
       expect(controller.state.optedIn).toBe(false);
-
-      messenger.call('AnalyticsController:optIn');
-
-      expect(controller.state.optedIn).toBe(true);
     });
+  });
 
-    it('handles optOut action', () => {
-      const { controller, messenger } = setupController({
-        state: { optedIn: true },
+  describe('stateChange event', () => {
```

Comment:
Is this test needed? If we are testing that the state changes when `optIn` is called then we are implicitly testing that the state change handler will be called. This is behavior that is common to all controllers, and is already tested in the tests for BaseController. So maybe we don't need this block?